--- lab2/kern/trap/trap.c	2020-02-25 07:00:17.772703736 -0800
+++ lab1/kern/trap/trap.c	2020-03-10 20:29:37.981094484 -0700
@@ -35,6 +35,12 @@
 void
 idt_init(void) {
      /* LAB1 YOUR CODE : STEP 2 */
+	extern uintptr_t __vectors[];
+	for(int i=0;i<256;i++){
+		SETGATE(idt[i], 0, GD_KTEXT, __vectors[i], DPL_KERNEL);	
+	}
+	lidt(&idt_pd);
+
      /* (1) Where are the entry addrs of each Interrupt Service Routine (ISR)?
       *     All ISR's entry addrs are stored in __vectors. where is uintptr_t __vectors[] ?
       *     __vectors[] is in kern/trap/vector.S which is produced by tools/vector.c
@@ -147,6 +153,10 @@
          * (2) Every TICK_NUM cycle, you can print some info using a funciton, such as print_ticks().
          * (3) Too Simple? Yes, I think so!
          */
+	ticks++;
+	if(ticks % TICK_NUM == 0){
+		print_ticks();
+	}
         break;
     case IRQ_OFFSET + IRQ_COM1:
         c = cons_getc();
